MNT: update dependency document (#1846)
IMAP-Science-Operations-Center/imap_processing: docs/source/data-access/data-dependency.rst

the L1A file, and therefore MAG L1B may be ready to begin processing.

Then, for each anticipated job, the batch starter process checks to see if all the upstream dependencies are met. Although we know we have one of the upstream dependencies for an expected job,
it's possible that there are other required dependencies that have not yet arrived. If we are missing required dependencies, then the system skip processing and waits for the missing files to arrive.

> Once missing file has arrived, it will trigger the same process, but all the required dependencies will be ready, and therefore processing job will be started.

Reviewer: ```suggestion
it's possible that there are other required dependencies that have not yet arrived. If we are missing any required dependencies, then the system does not kick off the processing job.
When the missing file arrives, it will trigger the same process of checking for all upstream dependencies. This time all required dependencies will be found and the processing job will be started.
```

- SPICE data type listed in the ``_SPICE_DIR_MAPPING`` dictionary in this file:
  `imap-data-access validation file <imap-data-validation_>`_

Primary Data Type
~~~~~~~~~~~~~~~~~~~~

Primary data type can be one of the following:

- IMAP data level listed in the ``VALID_DATALEVELS`` dictionary in this file:
  `imap-data-access Repo <imap-data-init_>`_

- ``spice``

- ``spin``

- ``repoint``

- ``ancillary``

Primary descriptor
~~~~~~~~~~~~~~~~~~~~

Primary descriptor can be one of the following:

- For science or ancillary data, the descriptors are defined by the instrument and SDC.

- For ``spice`` data types, ``historical`` and ``best`` are the valid descriptors.

- For ``spin`` and ``repoint`` data types, ``historical`` is the only valid descriptor.

Dependent Source
~~~~~~~~~~~~~~~~~~~

Same as primary_source, but for the dependent file.

Dependent Data Type
~~~~~~~~~~~~~~~~~~~~

Same as primary_data_type, but for the dependent file.

Dependent Descriptor
~~~~~~~~~~~~~~~~~~~~

Same as primary_descriptor, but for the dependent file.

Relationship
~~~~~~~~~~~~~~~~~~~

- **HARD**
  Triggers processing on file ingestion or a reprocessing event.

- **HARD_NO_TRIGGER**

>   Required data file, but it's assumed to already exist. It doesn't trigger
>   processing on file ingestion.

Reviewer 2: ```suggestion
  Required data file. However, a new version of this file doesn't trigger 
  processing on file ingestion.
```

AFAIK the batch starter still checks if the file exists, it's just that it won't start processing?
